File: Render/Renderer.cs
// 렌더러 레지스트리(플러그인) 스켈레톤.
namespace Dbgit.Render;

public enum OutputKind
{
    Text,
    Json,
    Markdown
}

public interface IRenderer
{
    OutputKind Kind { get; }
    string Render(string baseline, Dictionary<string, ProcDefinition> definitions);
}

public sealed class RendererRegistry
{
    private readonly Dictionary<OutputKind, IRenderer> _map = new();

    public RendererRegistry Register(IRenderer r)
    {
        _map[r.Kind] = r;
        return this;
    }

    public IRenderer Get(OutputKind kind)
    {
        if (_map.TryGetValue(kind, out var r))
            return r;
        throw new InvalidOperationException($"등록되지 않은 렌더러: {kind}");
    }
}

File: src/render/renderer.ts
/**
 * 렌더러 레지스트리(플러그인) 스켈레톤.
 *
 * 출력 포맷별 렌더러를 등록/조회합니다.
 */
import type { ProcDefinition } from "../compare.js";

export type RendererId = "text" | "json" | "markdown";

export interface Renderer {
  id: RendererId;
  render(baseline: string, definitions: Record<string, ProcDefinition>): string;
}

export class RendererRegistry {
  private readonly map = new Map<RendererId, Renderer>();

  register(r: Renderer): this {
    this.map.set(r.id, r);
    return this;
  }

  get(id: RendererId): Renderer {
    const r = this.map.get(id);
    if (!r) throw new Error(`등록되지 않은 렌더러: ${id}`);
    return r;
  }
}

